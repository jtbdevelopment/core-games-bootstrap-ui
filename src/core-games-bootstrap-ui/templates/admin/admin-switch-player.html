<div class="admin-user">
    <div class="row">
        <div class="col-xs-4">
            <button class="btn btn-default btn-default-focus btn-success btn-stop-simulating"
                    ng-disabled="!admin.revertEnabled" ng-click="admin.revertToNormal()">Stop
            </button>
        </div>
        <div class="col-xs-8">{{admin.revertText}}</div>
    </div>
    <div class="row">
        <div class="col-xs-4">
            <button class="btn btn-default btn-default-focus btn-primary btn-get-users"
                    ng-click="admin.refreshData()">Get Users
            </button>
        </div>
        <div class="col-xs-8">
            <input class="form-control" type="text" ng-model="admin.searchText"
                   placeholder="And Name contains..">
        </div>
    </div>
    <div class="row">
        <div class="col-xs-12">
            <table class="table table-striped table-condensed table-admin-users">
                <tr>
                    <th>Display Name</th>
                    <th>ID</th>
                    <th>Switch</th>
                </tr>
                <tr ng-repeat="player in admin.players">
                    <td>{{player.displayName}}</td>
                    <td>{{player.id}}</td>
                    <td>
                        <button class="btn btn-danger btn-danger btn-change-user"
                                ng-click="admin.switchToPlayer(player.id)">Switch to this user..
                        </button>
                    </td>
                </tr>
            </table>
        </div>
    </div>
    <div class="row">
        <div class="col-xs-12">
            <ul uib-pagination num-pages="admin.numberOfPages" items-per-page="admin.pageSize"
                direction-links="false" total-items="admin.totalItems" ng-model="admin.currentPage"
                ng-change="admin.changePage()"></ul>
        </div>
    </div>
</div>
