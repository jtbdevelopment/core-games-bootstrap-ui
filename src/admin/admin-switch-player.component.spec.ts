import {ComponentFixture, fakeAsync, inject, TestBed, tick} from '@angular/core/testing';
import {NgbModule, NgbPaginationConfig} from '@ng-bootstrap/ng-bootstrap';
import {AdminSwitchPlayerComponent} from './admin-switch-player.component';
import {FormsModule} from '@angular/forms';
import {Observable} from 'rxjs/Observable';
import {BehaviorSubject} from 'rxjs/BehaviorSubject';
import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
import {HttpRequest} from '@angular/common/http';
import {Player, PlayerService} from 'jtb-core-games-ui';

class MockPlayerService {
    static playerSubject: BehaviorSubject<Player> = new BehaviorSubject(new Player());
    static loggedInSubject: BehaviorSubject<Player> = new BehaviorSubject(new Player());

    player: Observable<Player> = Observable.from<Player>(MockPlayerService.playerSubject);
    loggedInPlayer: Observable<Player> = Observable.from<Player>(MockPlayerService.loggedInSubject);

    simulateUser = jasmine.createSpy('su');
}

describe('admin switch player component', () => {
    let httpMock: HttpTestingController;

    let initialPlayer = new Player({id: 'loggedin'});
    let simUser = new Player({id: 'sim'});

    let fixture: ComponentFixture<AdminSwitchPlayerComponent>;

    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                NgbModule,
                FormsModule,
                HttpClientTestingModule
            ],
            declarations: [
                AdminSwitchPlayerComponent
            ],
            providers: [
                {provide: PlayerService, useClass: MockPlayerService},
                NgbPaginationConfig
            ],
        });
        TestBed.compileComponents();
        httpMock = TestBed.get(HttpTestingController);
        fixture = TestBed.createComponent(AdminSwitchPlayerComponent);
        fixture.detectChanges();
        MockPlayerService.playerSubject.next(initialPlayer);
        MockPlayerService.loggedInSubject.next(initialPlayer);
    });

    it('should render results from http', () => {
        expect(fixture.componentInstance.revertEnabled).toBeFalsy();
        expect(fixture.componentInstance.revertText).toEqual('You are yourself.');
        expect(fixture.componentInstance.searchText).toEqual('');
    });

    describe('it after initial page of users loaded', () => {
        let players = [
            new Player({id: 'id1', displayName: 'dn1'}),
            new Player({id: 'id2', displayName: 'dn2'}),
            new Player({id: 'id3', displayName: 'dn4'}),
            new Player({id: 'id4', displayName: 'dn4'}),
        ];
        beforeEach(() => {
            let request = httpMock.expectOne(
                (req: HttpRequest<any>) => req.url === '/api/player/admin/playersLike' &&
                    req.params.get('pageSize') === '20' &&
                    req.params.get('page') === '0' &&
                    req.params.get('like') === ''
            );
            request.flush({
                totalElements: 40,
                number: 0,
                content: players
            });
            fixture.detectChanges();
        });

        afterEach(() => {
            httpMock.verify();
        });

        it('does render of players', () => {
            expect(fixture.componentInstance.currentPage).toBeCloseTo(1);
            expect(fixture.componentInstance.totalPlayers).toBeCloseTo(40);
            expect(JSON.stringify(fixture.componentInstance.players)).toEqual(JSON.stringify(players));

            players.forEach(p => {
                expect(fixture.nativeElement.querySelector('#' + p.id)).not.toBeNull();
            });
        });

        it('can change pages', fakeAsync(() => {
            let newPlayers = [
                new Player({id: 'id7', displayName: 'dn7'}),
                new Player({id: 'id8', displayName: 'dn8'}),
                new Player({id: 'id9', displayName: 'dn9'}),
                new Player({id: 'id0', displayName: 'dn0'}),
            ];

            fixture.componentInstance.currentPage = 2;
            fixture.detectChanges();
            tick();
            fixture.componentInstance.changePage();

            let request = httpMock.expectOne(
                (req: HttpRequest<any>) => req.url === '/api/player/admin/playersLike' &&
                    req.params.get('pageSize') === '20' &&
                    req.params.get('page') === '1' &&
                    req.params.get('like') === ''
            );
            request.flush({
                totalElements: 45,
                number: 1,
                content: newPlayers
            });
            fixture.detectChanges();
            expect(fixture.componentInstance.currentPage).toEqual(2);
            expect(fixture.componentInstance.totalPlayers).toBeCloseTo(45);
            expect(JSON.stringify(fixture.componentInstance.players)).toEqual(JSON.stringify(newPlayers));
            players.forEach(p => {
                expect(fixture.nativeElement.querySelector('#' + p.id)).toBeNull();
            });
            newPlayers.forEach(p => {
                expect(fixture.nativeElement.querySelector('#' + p.id)).not.toBeNull();
            });
        }));

        it('can change search text', () => {
            let newPlayers = [
                new Player({id: 'id7', displayName: 'dn7'})
            ];

            fixture.componentInstance.searchText = 'dn7';
            fixture.detectChanges();
            fixture.componentInstance.refreshUsers();

            let request = httpMock.expectOne(
                (req: HttpRequest<any>) => req.url === '/api/player/admin/playersLike' &&
                    req.params.get('pageSize') === '20' &&
                    req.params.get('page') === '0' &&
                    req.params.get('like') === 'dn7'
            );
            request.flush({
                totalElements: 1,
                number: 0,
                content: newPlayers
            });
            fixture.detectChanges();
            expect(fixture.componentInstance.currentPage).toEqual(1);
            expect(fixture.componentInstance.totalPlayers).toBeCloseTo(1);
            expect(JSON.stringify(fixture.componentInstance.players)).toEqual(JSON.stringify(newPlayers));

            players.forEach(p => {
                expect(fixture.nativeElement.querySelector('#' + p.id)).toBeNull();
            });
            newPlayers.forEach(p => {
                expect(fixture.nativeElement.querySelector('#' + p.id)).not.toBeNull();
            });
        });

        it('switching players', inject([PlayerService], (playerService) => {
            fixture.componentInstance.switchToPlayer(simUser.id);
            expect(playerService.simulateUser).toHaveBeenCalledWith(simUser.id);
            MockPlayerService.playerSubject.next(simUser);
            fixture.detectChanges();

            expect(fixture.componentInstance.revertEnabled).toBeTruthy();
            expect(fixture.componentInstance.revertText).toEqual('You are simulating another player.');
        }));

        it('reverting', inject([PlayerService], (playerService) => {
            fixture.componentInstance.revertEnabled = true;
            fixture.componentInstance.revertText = '';
            fixture.componentInstance.revertToNormal();
            expect(playerService.simulateUser).toHaveBeenCalledWith(initialPlayer.id);
            MockPlayerService.playerSubject.next(initialPlayer);
            fixture.detectChanges();

            expect(fixture.componentInstance.revertEnabled).toBeFalsy();
            expect(fixture.componentInstance.revertText).toEqual('You are yourself.');
        }));
    });

});
