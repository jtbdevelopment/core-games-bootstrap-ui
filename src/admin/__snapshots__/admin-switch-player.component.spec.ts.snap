// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin switch player component it after initial page of users loaded can change pages 1`] = `
<admin-switch-player
  currentPage={[Function Number]}
  http={[Function HttpClient]}
  loggedInPlayer={[Function Player]}
  pageSize={[Function Number]}
  player={[Function Player]}
  playerService={[Function MockPlayerService]}
  players={[Function Array]}
  revertEnabled="false"
  revertText={[Function String]}
  searchText=""
  totalPlayers={[Function Number]}
>
  <div
    class="admin-user"
  >
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col-4"
      >
        
              
        <button
          class="btn btn-secondary btn-success btn-stop-simulating"
          disabled=""
        >
          Stop
              
        </button>
        
          
      </div>
      
          
      <div
        class="col-8"
      >
        You are yourself.
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col-4"
      >
        
              
        <button
          class="btn btn-default focus btn-primary btn-get-users"
        >
          Get Users
              
        </button>
        
          
      </div>
      
          
      <div
        class="col-8"
      >
        
              
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          placeholder="And Name contains.."
          type="text"
        />
        
          
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col"
      >
        
              
        <table
          class="table table-striped table-sm table-admin-users"
        >
          
                  
          <tbody>
            <tr>
              
                      
              <th>
                Display Name
              </th>
              
                      
              <th>
                ID
              </th>
              
                      
              <th>
                Switch
              </th>
              
                  
            </tr>
            
                  
            
            <tr>
              
                      
              <td>
                dn7
              </td>
              
                      
              <td>
                id7
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id7"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            <tr>
              
                      
              <td>
                dn8
              </td>
              
                      
              <td>
                id8
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id8"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            <tr>
              
                      
              <td>
                dn9
              </td>
              
                      
              <td>
                id9
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id9"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            <tr>
              
                      
              <td>
                dn0
              </td>
              
                      
              <td>
                id0
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id0"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            
              
          </tbody>
        </table>
        
          
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col"
      >
        
              
        <ngb-pagination
          ng-reflect-collection-size="45"
          ng-reflect-direction-links="false"
          ng-reflect-page="2"
          ng-reflect-page-size="20"
          role="navigation"
        >
          
      
          <ul
            class="pagination"
          >
            
        
            
            
  
        
            
            
        
            
            <li
              class="page-item"
            >
              
          
              
              
          
              
              <a
                class="page-link"
                href=""
              >
                
            1
            
                
                
          
              </a>
              
        
            </li>
            <li
              class="page-item active"
            >
              
          
              
              
          
              
              <a
                class="page-link"
                href=""
              >
                
            2
            
                
                <span
                  class="sr-only"
                >
                  (current)
                </span>
                
          
              </a>
              
        
            </li>
            <li
              class="page-item"
            >
              
          
              
              
          
              
              <a
                class="page-link"
                href=""
              >
                
            3
            
                
                
          
              </a>
              
        
            </li>
            
        
            
            
  
        
            
            
      
          </ul>
          
    
        </ngb-pagination>
        
          
      </div>
      
      
    </div>
    
  
  </div>
  
</admin-switch-player>
`;

exports[`admin switch player component it after initial page of users loaded can change search text 1`] = `
<admin-switch-player
  currentPage={[Function Number]}
  http={[Function HttpClient]}
  loggedInPlayer={[Function Player]}
  pageSize={[Function Number]}
  player={[Function Player]}
  playerService={[Function MockPlayerService]}
  players={[Function Array]}
  revertEnabled="false"
  revertText={[Function String]}
  searchText={[Function String]}
  totalPlayers={[Function Number]}
>
  <div
    class="admin-user"
  >
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col-4"
      >
        
              
        <button
          class="btn btn-secondary btn-success btn-stop-simulating"
          disabled=""
        >
          Stop
              
        </button>
        
          
      </div>
      
          
      <div
        class="col-8"
      >
        You are yourself.
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col-4"
      >
        
              
        <button
          class="btn btn-default focus btn-primary btn-get-users"
        >
          Get Users
              
        </button>
        
          
      </div>
      
          
      <div
        class="col-8"
      >
        
              
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model="dn7"
          placeholder="And Name contains.."
          type="text"
        />
        
          
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col"
      >
        
              
        <table
          class="table table-striped table-sm table-admin-users"
        >
          
                  
          <tbody>
            <tr>
              
                      
              <th>
                Display Name
              </th>
              
                      
              <th>
                ID
              </th>
              
                      
              <th>
                Switch
              </th>
              
                  
            </tr>
            
                  
            
            <tr>
              
                      
              <td>
                dn7
              </td>
              
                      
              <td>
                id7
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id7"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            
              
          </tbody>
        </table>
        
          
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col"
      >
        
              
        <ngb-pagination
          ng-reflect-collection-size="1"
          ng-reflect-direction-links="false"
          ng-reflect-page="1"
          ng-reflect-page-size="20"
          role="navigation"
        >
          
      
          <ul
            class="pagination"
          >
            
        
            
            
  
        
            
            
        
            
            <li
              class="page-item active"
            >
              
          
              
              
          
              
              <a
                class="page-link"
                href=""
              >
                
            1
            
                
                <span
                  class="sr-only"
                >
                  (current)
                </span>
                
          
              </a>
              
        
            </li>
            
        
            
            
  
        
            
            
      
          </ul>
          
    
        </ngb-pagination>
        
          
      </div>
      
      
    </div>
    
  
  </div>
  
</admin-switch-player>
`;

exports[`admin switch player component it after initial page of users loaded does render of players 1`] = `
<admin-switch-player
  currentPage={[Function Number]}
  http={[Function HttpClient]}
  loggedInPlayer={[Function Player]}
  pageSize={[Function Number]}
  player={[Function Player]}
  playerService={[Function MockPlayerService]}
  players={[Function Array]}
  revertEnabled="false"
  revertText={[Function String]}
  searchText=""
  totalPlayers={[Function Number]}
>
  <div
    class="admin-user"
  >
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col-4"
      >
        
              
        <button
          class="btn btn-secondary btn-success btn-stop-simulating"
          disabled=""
        >
          Stop
              
        </button>
        
          
      </div>
      
          
      <div
        class="col-8"
      >
        You are yourself.
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col-4"
      >
        
              
        <button
          class="btn btn-default focus btn-primary btn-get-users"
        >
          Get Users
              
        </button>
        
          
      </div>
      
          
      <div
        class="col-8"
      >
        
              
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          placeholder="And Name contains.."
          type="text"
        />
        
          
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col"
      >
        
              
        <table
          class="table table-striped table-sm table-admin-users"
        >
          
                  
          <tbody>
            <tr>
              
                      
              <th>
                Display Name
              </th>
              
                      
              <th>
                ID
              </th>
              
                      
              <th>
                Switch
              </th>
              
                  
            </tr>
            
                  
            
            <tr>
              
                      
              <td>
                dn1
              </td>
              
                      
              <td>
                id1
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id1"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            <tr>
              
                      
              <td>
                dn2
              </td>
              
                      
              <td>
                id2
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id2"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            <tr>
              
                      
              <td>
                dn4
              </td>
              
                      
              <td>
                id3
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id3"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            <tr>
              
                      
              <td>
                dn4
              </td>
              
                      
              <td>
                id4
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id4"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            
              
          </tbody>
        </table>
        
          
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col"
      >
        
              
        <ngb-pagination
          ng-reflect-collection-size="40"
          ng-reflect-direction-links="false"
          ng-reflect-page="1"
          ng-reflect-page-size="20"
          role="navigation"
        >
          
      
          <ul
            class="pagination"
          >
            
        
            
            
  
        
            
            
        
            
            <li
              class="page-item active"
            >
              
          
              
              
          
              
              <a
                class="page-link"
                href=""
              >
                
            1
            
                
                <span
                  class="sr-only"
                >
                  (current)
                </span>
                
          
              </a>
              
        
            </li>
            <li
              class="page-item"
            >
              
          
              
              
          
              
              <a
                class="page-link"
                href=""
              >
                
            2
            
                
                
          
              </a>
              
        
            </li>
            
        
            
            
  
        
            
            
      
          </ul>
          
    
        </ngb-pagination>
        
          
      </div>
      
      
    </div>
    
  
  </div>
  
</admin-switch-player>
`;

exports[`admin switch player component it after initial page of users loaded reverting 1`] = `
<admin-switch-player
  currentPage={[Function Number]}
  http={[Function HttpClient]}
  loggedInPlayer={[Function Player]}
  pageSize={[Function Number]}
  player={[Function Player]}
  playerService={[Function MockPlayerService]}
  players={[Function Array]}
  revertEnabled="false"
  revertText={[Function String]}
  searchText=""
  totalPlayers={[Function Number]}
>
  <div
    class="admin-user"
  >
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col-4"
      >
        
              
        <button
          class="btn btn-secondary btn-success btn-stop-simulating"
          disabled=""
        >
          Stop
              
        </button>
        
          
      </div>
      
          
      <div
        class="col-8"
      >
        You are yourself.
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col-4"
      >
        
              
        <button
          class="btn btn-default focus btn-primary btn-get-users"
        >
          Get Users
              
        </button>
        
          
      </div>
      
          
      <div
        class="col-8"
      >
        
              
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          placeholder="And Name contains.."
          type="text"
        />
        
          
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col"
      >
        
              
        <table
          class="table table-striped table-sm table-admin-users"
        >
          
                  
          <tbody>
            <tr>
              
                      
              <th>
                Display Name
              </th>
              
                      
              <th>
                ID
              </th>
              
                      
              <th>
                Switch
              </th>
              
                  
            </tr>
            
                  
            
            <tr>
              
                      
              <td>
                dn1
              </td>
              
                      
              <td>
                id1
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id1"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            <tr>
              
                      
              <td>
                dn2
              </td>
              
                      
              <td>
                id2
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id2"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            <tr>
              
                      
              <td>
                dn4
              </td>
              
                      
              <td>
                id3
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id3"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            <tr>
              
                      
              <td>
                dn4
              </td>
              
                      
              <td>
                id4
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id4"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            
              
          </tbody>
        </table>
        
          
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col"
      >
        
              
        <ngb-pagination
          ng-reflect-collection-size="40"
          ng-reflect-direction-links="false"
          ng-reflect-page="1"
          ng-reflect-page-size="20"
          role="navigation"
        >
          
      
          <ul
            class="pagination"
          >
            
        
            
            
  
        
            
            
        
            
            <li
              class="page-item active"
            >
              
          
              
              
          
              
              <a
                class="page-link"
                href=""
              >
                
            1
            
                
                <span
                  class="sr-only"
                >
                  (current)
                </span>
                
          
              </a>
              
        
            </li>
            <li
              class="page-item"
            >
              
          
              
              
          
              
              <a
                class="page-link"
                href=""
              >
                
            2
            
                
                
          
              </a>
              
        
            </li>
            
        
            
            
  
        
            
            
      
          </ul>
          
    
        </ngb-pagination>
        
          
      </div>
      
      
    </div>
    
  
  </div>
  
</admin-switch-player>
`;

exports[`admin switch player component it after initial page of users loaded switching players 1`] = `
<admin-switch-player
  currentPage={[Function Number]}
  http={[Function HttpClient]}
  loggedInPlayer={[Function Player]}
  pageSize={[Function Number]}
  player={[Function Player]}
  playerService={[Function MockPlayerService]}
  players={[Function Array]}
  revertEnabled={[Function Boolean]}
  revertText={[Function String]}
  searchText=""
  totalPlayers={[Function Number]}
>
  <div
    class="admin-user"
  >
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col-4"
      >
        
              
        <button
          class="btn btn-secondary btn-success btn-stop-simulating"
        >
          Stop
              
        </button>
        
          
      </div>
      
          
      <div
        class="col-8"
      >
        You are simulating another player.
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col-4"
      >
        
              
        <button
          class="btn btn-default focus btn-primary btn-get-users"
        >
          Get Users
              
        </button>
        
          
      </div>
      
          
      <div
        class="col-8"
      >
        
              
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          placeholder="And Name contains.."
          type="text"
        />
        
          
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col"
      >
        
              
        <table
          class="table table-striped table-sm table-admin-users"
        >
          
                  
          <tbody>
            <tr>
              
                      
              <th>
                Display Name
              </th>
              
                      
              <th>
                ID
              </th>
              
                      
              <th>
                Switch
              </th>
              
                  
            </tr>
            
                  
            
            <tr>
              
                      
              <td>
                dn1
              </td>
              
                      
              <td>
                id1
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id1"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            <tr>
              
                      
              <td>
                dn2
              </td>
              
                      
              <td>
                id2
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id2"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            <tr>
              
                      
              <td>
                dn4
              </td>
              
                      
              <td>
                id3
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id3"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            <tr>
              
                      
              <td>
                dn4
              </td>
              
                      
              <td>
                id4
              </td>
              
                      
              <td>
                
                          
                <button
                  class="btn btn-danger btn-danger btn-change-user"
                  id="id4"
                >
                  Switch to this user..
                          
                </button>
                
                      
              </td>
              
                  
            </tr>
            
              
          </tbody>
        </table>
        
          
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col"
      >
        
              
        <ngb-pagination
          ng-reflect-collection-size="40"
          ng-reflect-direction-links="false"
          ng-reflect-page="1"
          ng-reflect-page-size="20"
          role="navigation"
        >
          
      
          <ul
            class="pagination"
          >
            
        
            
            
  
        
            
            
        
            
            <li
              class="page-item active"
            >
              
          
              
              
          
              
              <a
                class="page-link"
                href=""
              >
                
            1
            
                
                <span
                  class="sr-only"
                >
                  (current)
                </span>
                
          
              </a>
              
        
            </li>
            <li
              class="page-item"
            >
              
          
              
              
          
              
              <a
                class="page-link"
                href=""
              >
                
            2
            
                
                
          
              </a>
              
        
            </li>
            
        
            
            
  
        
            
            
      
          </ul>
          
    
        </ngb-pagination>
        
          
      </div>
      
      
    </div>
    
  
  </div>
  
</admin-switch-player>
`;

exports[`admin switch player component should render results from http 1`] = `
<admin-switch-player
  currentPage={[Function Number]}
  http={[Function HttpClient]}
  loggedInPlayer={[Function Player]}
  pageSize={[Function Number]}
  player={[Function Player]}
  playerService={[Function MockPlayerService]}
  players={[Function Array]}
  revertEnabled="false"
  revertText={[Function String]}
  searchText=""
  totalPlayers="0"
>
  <div
    class="admin-user"
  >
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col-4"
      >
        
              
        <button
          class="btn btn-secondary btn-success btn-stop-simulating"
          disabled=""
        >
          Stop
              
        </button>
        
          
      </div>
      
          
      <div
        class="col-8"
      >
        You are yourself.
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col-4"
      >
        
              
        <button
          class="btn btn-default focus btn-primary btn-get-users"
        >
          Get Users
              
        </button>
        
          
      </div>
      
          
      <div
        class="col-8"
      >
        
              
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          placeholder="And Name contains.."
          type="text"
        />
        
          
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col"
      >
        
              
        <table
          class="table table-striped table-sm table-admin-users"
        >
          
                  
          <tbody>
            <tr>
              
                      
              <th>
                Display Name
              </th>
              
                      
              <th>
                ID
              </th>
              
                      
              <th>
                Switch
              </th>
              
                  
            </tr>
            
                  
            
            
              
          </tbody>
        </table>
        
          
      </div>
      
      
    </div>
    
      
    <div
      class="row"
    >
      
          
      <div
        class="col"
      >
        
              
        <ngb-pagination
          ng-reflect-collection-size="0"
          ng-reflect-direction-links="false"
          ng-reflect-page="1"
          ng-reflect-page-size="20"
          role="navigation"
        >
          
      
          <ul
            class="pagination"
          >
            
        
            
            
  
        
            
            
        
            
            
        
            
            
  
        
            
            
      
          </ul>
          
    
        </ngb-pagination>
        
          
      </div>
      
      
    </div>
    
  
  </div>
  
</admin-switch-player>
`;
